src/centralserver/admin-service/src/main/java/org/niis/xroad/centralserver/restapi/service/ClientSearchService.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * The MIT License
+ *
+ * Copyright (c) 2019- Nordic Institute for Interoperability Solutions (NIIS)
+ * Copyright (c) 2018 Estonian Information System Authority (RIA),
+ * Nordic Institute for Interoperability Solutions (NIIS), Population Register Centre (VRK)
+ * Copyright (c) 2015-2017 Estonian Information System Authority (RIA), Population Register Centre (VRK)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.niis.xroad.centralserver.restapi.service;
+
+import lombok.RequiredArgsConstructor;
+import org.niis.xroad.centralserver.restapi.dto.FlattenedSecurityServerClientDto;
+import org.niis.xroad.centralserver.restapi.repository.FlattenedSecurityServerClientRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+
+import java.util.stream.Collectors;
+
+/**
+ * Service for searching {@link org.niis.xroad.centralserver.restapi.entity.FlattenedSecurityServerClient}s
+ */
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class ClientSearchService {
+    @Autowired
+    private FlattenedSecurityServerClientRepository flattenedClientRepository;
+
+    public Page<FlattenedSecurityServerClientDto> find(
+            FlattenedSecurityServerClientRepository.SearchParameters params,
+            Pageable pageable) {
+        var clients = flattenedClientRepository.findAll(
```

Comment:
Needs to add a secondary sort parameter that guarantees stable sorting and paging. See https://github.com/nordic-institute/X-Road/pull/1207

Reply:
Addressed in https://github.com/nordic-institute/X-Road/commit/3b247a58e486f7942cddbf568eb19ff6198ac886